### PR TITLE
Better stack trace

### PR DESCRIPTION
**Pull request by Vijar**
@koulmomo

When I was debugging https://github.com/yahoo/fluxible-plugin-fetchr/issues/46 the stack trace wasn't very helpful and didn't point out which line in `fetchr` the error came from. This PR fixes it and will even work in IE.

Before:
screen shot 2015-09-03 at 1 35 35 pm

After:
screen shot 2015-09-03 at 1 34 44 pm

### CODE REVIEW

File: HttpError.js [hunk 1]
```diff
@@ -22,12 +22,22 @@ function HttpError (status, message, options) {
     if (options.debug) {
         this.debug = options.debug;
     }
+
+    // Stack Trace
+    if(Error.captureStackTrace) {// AKA if not IE
```
Comment (koulmomo) [inline HttpError.js]: nitpick, don't inline comment, move it right above or right below the if statement.

File: HttpError.js [hunk 2]
```diff
@@ -22,12 +22,22 @@ function HttpError (status, message, options) {
...
+        Error.captureStackTrace(this, this.constructor)
+    } else {
+        // In IE
+        this.stack = (new Error()).stack;
+        this.toString = function () {
```
Comment (koulmomo) [inline HttpError.js]: I don't think this part is necessary.

**Comment by koulmomo** [comment]
bump patch and :shipit:

I don't have merge permissions anymore, so I think you need @akshayp 

I'll tag and release once it's in

File: HttpError.js (hunk 2 above)
Comment (Vijar) [inline HttpError.js]: I thought I'd include it for IE people...but yeah, who cares/

File: HttpError.js [hunk 3]
```diff
@@ -22,12 +22,20 @@ function HttpError (status, message, options) {
     if (options.debug) {
         this.debug = options.debug;
     }
+
+    // Stack Trace
+    if(Error.captureStackTrace) {
```
Comment (koulmomo) [inline HttpError.js]: lint error, whitespace after `if`

**Comment by Vijar** [comment]
@koulmomo I also realized I can get an even better stack trace if I throw HttpError myself so I added it to index.js exports.
screen shot 2015-09-04 at 10 46 35 am

**Comment by Vijar** [comment]
So instead of 
```
var fumble = require('fumble');
var error = fumble.http.badRequest('Invalid Fetchr Access');
```
I'm just going to do this:
```
var HttpError = require('fumble').HttpError;
var error = new HttpError(400, 'Invalid Fetchr Access');
```

**Comment by koulmomo** [comment]
@Vijar sounds good, you can always do `var HttpError = require('fumble/HttpError')` btw to not require all of the http methods if you only ever use `HttpError`.

If you have the time, I'm wondering if it would be worth modifying the `stack` string on `HttpError` creation, since the benefit of the lib is the fact that you can just do `.badRequest()`, `.notFound()`, etc...

so, right after creating the error in `create`, modify the stack.
```js
var err = new HttpError(status, message, opts);

 // this is the index of the newline after: HttpError: message
var firstNewlineIndex = err.stack.indexOf('\n');

// this is the index of the newline after "at create ...."
var secondNewlineIndex = err.stack.indexOf('\n', firstNewline+1);

// this is the index of the newline after "at Object.badRequest", this may or may not exist, depending on whether create was called directly (fumble.http.create) vs indirectly (fumble.http.badRequest)
var lastNewLineIndex = err.stack.indexOf('\n', err.stack.indexOf('fumble', secondNewline));

// create was called directly 
if (lastNewline < 0) {
lastNewlineIndex = secondNewlineIndex;
}
err.stack = err.stack.substring(0, firstNewline+1) + err.stack.substring(lastNewlineIndex+1);
```
^ this of course needs test, and possibly a quick `jsperf` test for the last part to see how expensive `substring` + concat vs other options.

thoughts?

**Comment by Vijar** [comment]
@koulmomo that sounds error prone, but I can play around with it in a different PR.

**Pull request merged**